packages/react/src/hooks/useGetUrl.ts

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+
+import { getUrl, GetUrlInput } from 'aws-amplify/storage';
+
+import { isFunction } from '@aws-amplify/ui';
+import { useHasValueUpdated } from '@aws-amplify/ui-react-core';
+
+export type UseGetUrlInput = GetUrlInput & {
+  onError?: (error: Error) => void;
```

Comment:
Initially tried interfaces over types but get this ts error `An interface can only extend an object type or intersection of object types with statically known members.`